src/java/org/apache/cassandra/cql3/restrictions/SimpleRestriction.java

```diff
@@ -327,8 +327,11 @@ public void addToRowFilter(RowFilter filter, IndexRegistry indexRegistry, QueryO
         {
             case SINGLE_COLUMN:
                 List<ByteBuffer> buffers = bindAndGet(options);
-                if (operator == Operator.IN || operator == Operator.BETWEEN)
+                if (operator.kind() == Operator.Kind.TERNARY || operator.kind() == Operator.Kind.MULTI_VALUE)
                 {
+                    // For BETWEEN we support like in SQL reversed bounds
+                    if (operator.kind() == Operator.Kind.TERNARY)
```

Comment:
```suggestion
                    if (column.isClusteringColumn())
```
I was wondering if we can do something like this to avoid sorting when we don't have a clustering key column in the first place?

Reply:
The sort is always required because we support like in SQL between with reversed bounds. For example `BETWEEN 1 AND 2` and `BETWEEN 2 AND 1` are both valid and will return the same rows.